### ISSUE TITLE
oCap MVP: Logged out view

### ISSUE THREAD

**Issue by bdresser** [MetaMask/Design#114]
When the user is not connected to a dapp and they open the extension, what view do they see? More broadly, how should we shift the representation of accounts and the 1:1 relationship of account:extension once we have multiple accounts simultaneously connected to multiple dapps?

This is part of oCaps MVP and relates to https://github.com/MetaMask/Design/issues/58

**Comment by cjeria** [comment]
Some preliminary sketches / ideas to kick start the discussion. This has a lot of potentially big implications for the behavior of the extension. 

Might be easier to schedule a design jam vs trying to do this async to get everyone's ideas out there on how this could work!
cc @danfinlay @omnat 

![image](https://user-images.githubusercontent.com/597770/61092733-17a91d80-a40d-11e9-915e-9df96f77ad06.png)

**Comment by danfinlay** [comment]
That looks good to me, Christian! It's nice and simple!

I'm a little confused on the accounts/identities distinction here.

Also I think we might want to make it very clear whether clicking one of these accounts will _view_ it, or _log in with that account_.

We might even want to make a page before the account list, one that distinguishes:
- Log in to [site name] (opens account list)
- Open MetaMask Settings (opens full screen mode).

**Comment by omnat** [comment]
Looking good! 

I am also curious about the account - identity distinction. Is identity related too permissions from this specific account given to specific dapps?

A jam would indeed be great for this one!

**Comment by omnat** [comment]
https://docs.google.com/presentation/d/1vOEGNARZwcXWAGvnpzSLP97dFvYALuOgNE5PfKxTNfE/edit#slide=id.p
Presented by Christian in design sprint sync of Sprint 16

**Comment by danfinlay** [comment]
@omnat could you add those notes from the end of our last meeting here? The ones where you said you'd like feedback on it?

**Comment by omnat** [comment]
Yes, thanks for pinging about it @danfinlay ! There were some open questions I wanted to discuss at some point. 

- Section I is more relevant and specific to current focus,

- Section II is a bit heavy/unsuitable for Github IMO so dropping a Google doc link. This could be part of a different in-sync session or the Google doc for in-line commenting / editing / async discussion. Wdyt?


### Section I: Questions relevant to this sprint's permissions design work 
- What information is most important for users to see when a permission is asked?
E.g., Clear permission, money asked, give context on why needed (per case)
- Are some permissions _required_ to utilize functionality of this dapp? And others optional?
- Are all revokable permissions?
- What information is most important for users to see per dapp in settings?
E.g., Google shows the list of permissions, and when permission was granted

### Section II: General questions about permissions system:
For async discussion here: comment / edit - whatever works!
https://docs.google.com/document/d/1JfK5WDDsAIuUDScUAAb-A4E5vE887-QWblRyMcoJD_4/edit?usp=sharing

**Comment by danfinlay** [comment]
> What information is most important for users to see when a permission is asked?

I think this varies depending on the type of permission. Different permissions might have different kinds of _attenuation_: Account permission includes account selection, and maybe a logout timer. A spending permission probably includes an explicit limit. A signature challenge permission may have a time frame, or a specific parameter involved. So we probably need to discuss & design each permission we add, but this is worth it, because each permission could give applications many new abilities.

> Are some permissions required to utilize functionality of this dapp? And others optional?

I think it makes a lot of sense to give dapps some ability to express their needs. This can include the ability to flag a permission as mandatory, which we might disallow deselecting, forcing a user to fully reject the confirmation to avoid, or we might also allow apps to provide _context_ for why they are requesting each permission.

> Are all revokable permissions?

Yes, I think all permissions should be revocable, but some revocations may eventually require an on-chain transaction. If it wasn't revocable, it wouldn't be a permission, it would be a transfer of rights.

> What information is most important for users to see per dapp in settings?

Some guesses:
- The registered domain
- The account logged in
- The time first logged in?
- Maybe a log of activity on that site?
- The permissions granted to that site, and the ability to revoke them.
- A log out button.

**Comment by omnat** [comment]
Per design sprint planning - for this sprint the steps are:
- Send out the small survey on how people use accounts
- Create mocks to be tested (check inspiration for Accounts/Identity from Google Pay, Apple Pay, PayTM, BHIM app, on identity+wallet concept representation)
- This needs to be aligned with the Connected dapps wireframes Rachel made
- Create low/mid-fi fidelity versions for 3 oCap issues to do user testing for how this will work holistically? : Scope is to figure the navigation structure between all accounts / logged-out view, and permissions management of the connected account. 

Next sprint:
- Omna to do user testing 30 minutes 1:1 for the complete oCap

**Comment by omnat** [comment]
Survey results in detail here: https://docs.google.com/presentation/d/11A0Mvl9bNXGF2_GAkhyI6_PNSBN48zzuszm1msjLfkI/edit?usp=sharing

Highlights or main insights and recommendations for design:

<img width="1001" alt="Screen Shot 2019-08-12 at 10 43 13 AM" src="https://user-images.githubusercontent.com/44038140/62885569-0db95980-bcee-11e9-8d15-2647efc63cfc.png">

<img width="999" alt="Screen Shot 2019-08-12 at 10 43 32 AM" src="https://user-images.githubusercontent.com/44038140/62885568-0d20c300-bcee-11e9-8fde-568aea53daed.png">

**Comment by bdresser** [comment]
One thing I noticed during the survey - I keep track of my accounts by account name, but i also use all default names (account 1, 2, 3). Responses that just indicate they use names don't automatically mean **custom** names!

**Comment by omnat** [comment]
Good point Bobby! Per interviews in the past, I've heard some users deliberately have a naming convention that works for themselves. Could consider it as an emerged best practice (ish). In the past, even some active users has asked this as a feature request "I wish I could change the name of the account". This indicated that the edit name features (in extension) isn't as discoverable, and people could benefit from it. Especially as we move towards more mindfully choosing accounts/identities. So making account naming a more prominent part of some flow (maybe new account creation?) would be useful.

**Comment by cjeria** [comment]
@omnat did you mean to close this issue? I’m still working on it.

**Comment by omnat** [comment]
Ah my bad. I mistakenly did 'close and comment' it seems.

**Comment by omnat** [comment]
Sprint 19:
- @rachelcope to take the 1st stab at stitching of the 3 oCap prototypes ready for testing (tasks in the last page of : https://docs.google.com/document/d/1zeaemPei0TnWGkkJaqbvP5j38hXfOQANfa2-RLmXpxI/edit#)
- consider making dapp-primary brand focus in the requests per feedback of making transaction requests contextual to dapp. past exploration: https://www.figma.com/file/e4KriNtpo80qONJMULQtCLDS/Research-outcomes?node-id=51%3A26
- @cjeria @rachelcope to review before testing
- user test complete oCap (this week @omnat )
